builtin/credential/appgroup/path_app.go

```diff
@@ -0,0 +1,931 @@
+package appgroup
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/helper/policyutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+// appStorageEntry stores all the options that are set on an App
+type appStorageEntry struct {
+	// Policies that are to be required by the token to access the App
+	Policies []string `json:"policies" structs:"policies" mapstructure:"policies"`
+
+	// Number of times the UserID generated against the App can be used to perform login
+	NumUses int `json:"num_uses" structs:"num_uses" mapstructure:"num_uses"`
+
+	// Duration (less than the backend's mount) after which a UserID generated against the App will expire
```

Comment:
"...less than the backend mount's max TTL" is I assume what you mean here? Just for clarity.